contrib/lib/src/ron.rs

```diff
@@ -0,0 +1,189 @@
+/// instance, to increase the RON limit to 5MiB for all environments, you may
+/// add the following to your `Rocket.toml`:
+///
+/// ```toml
+/// [global.limits]
+/// ron = 5242880
+/// ```
+#[derive(Debug)]
+pub struct Ron<T>(pub T);
+
+impl<T> Ron<T> {
+    /// Consumes the RON wrapper and returns the wrapped item.
+    ///
+    /// # Example
+    /// ```rust
+    /// # use rocket_contrib::ron::Ron;
+    /// let string = "Hello".to_string();
+    /// let my_ron = Ron(string);
+    /// assert_eq!(my_ron.into_inner(), "Hello".to_string());
+    /// ```
+    #[inline(always)]
+    pub fn into_inner(self) -> T {
+        self.0
+    }
+}
+
+
+/// Default limit for RON is 1MB.
+const LIMIT: u64 = 1 << 20;
+
+/// An error returned by the [`Ron`] data guard when incoming data fails to
+/// serialize as RON.
+#[derive(Debug)]
+pub enum RonError<'a> {
+    /// An I/O error occurred while reading the incoming request data.
+    Io(io::Error),
+
+    /// The client's data was received successfully but failed to parse as valid
+    /// RON or as the requested type. The `&str` value in `.0` is the raw data
+    /// received from the user, while the `Error` in `.1` is the deserialization
+    /// error from `serde`.
+    Parse(&'a str, ron_crate::de::Error),
+}
+
+impl<'a, T: Deserialize<'a>> FromData<'a> for Ron<T> {
+    type Error = RonError<'a>;
+    type Owned = String;
+    type Borrowed= str;
+ 
+    fn transform<'r>(r: &'r Request<'_>, d: Data) -> TransformFuture<'r, Self::Owned, Self::Error> {
+        let size_limit = r.limits().get("ron").unwrap_or(LIMIT);
+        Box::pin(async move {
+            let mut s = String::with_capacity(512);
+            let mut reader = d.open().take(size_limit);
+            match reader.read_to_string(&mut s).await {
+                Ok(_) => Borrowed(Success(s)),
+                Err(e) => Borrowed(Failure((Status::BadRequest, RonError::Io(e))))
+            }
+        })
+    }
+
+    fn from_data(_: &Request<'_>, o: Transformed<'a, Self>) -> FromDataFuture<'a, Self, Self::Error> {
+        Box::pin(async move {
+            let string = try_outcome!(o.borrowed());
+            match ron_crate::de::from_str(&string) {
+                Ok(v) => Success(Ron(v)),
+                Err(e) => {
+                    error_!("Couldn't parse RON body: {:?}", e);
+                    Failure((Status::BadRequest, RonError::Parse(string, e)))
+                }
+            }
+        })
+    }
+}
+
+
+/// Serializes the wrapped value into RON. Returns a response with Content-Type
+/// Text and a fixed-size body with the serialized value. If serialization
+/// fails, an `Err` of `Status::InternalServerError` is returned.
+impl<'r, T: Serialize> Responder<'r> for Ron<T> {
+    fn respond_to<'a, 'x>(self, req: &'r Request<'a>) -> BoxFuture<'x, response::Result<'r>>
+        where 'a: 'x, 'r: 'x, Self: 'x
+    {
+        match ron_crate::ser::to_string_pretty(&self.0, ron_crate::ser::PrettyConfig::default()) {
```

Comment:
Somewhere along the way we lost struct names. As far as I can tell this does not matter for deserialization, but we should probably either keep them for readability, or use entirely non-pretty output if speed/size is preferred.

Reply:
It now serializes correctly with struct names. And yes, the struct names are completely optional for ron.